src/main/java/Tree/AVLTree.java

```diff
@@ -144,36 +146,37 @@ public void delete(int delete_key) {
                     int parent_key = parent_node.getKey();
                     if (delete_key < parent_key) {
                         parent_node.setLeftNode(null);
-                        updateHeightAfterDeletion(parent_node);
+                        nodeToUpdate = parent_node;
                     } else {
                         parent_node.setRightNode(null);
-                        updateHeightAfterDeletion(parent_node);
+                        nodeToUpdate = parent_node;
                     }
                 }
             } else {
                 swapWithPredecessor(deleteNode, predecessor);
-                updateHeightAfterDeletion((AVLNode<T>) predecessor.getParentNode());
+                nodeToUpdate = (AVLNode<T>) predecessor.getParentNode();
             }
         } else {
             swapWithSuccessor(deleteNode, successor);
-            updateHeightAfterDeletion((AVLNode<T>) successor.getParentNode());
+            nodeToUpdate = (AVLNode<T>) successor.getParentNode();
         }
+        updateHeightAfterDeletion(nodeToUpdate);
     }
 
     protected void rightRotation() {
         AVLNode<T> newRoot = (AVLNode<T>) current.getLeftNode();
 
         if (newRoot.getRightNode() != null) {
-            current.setLeftNode(newRoot.getRightNode());
+            linkNodes(current, newRoot.getRightNode(), 'L');
         } else {
-            current.setLeftNode(null);
+            linkNodes(current, null, 'L');
         }
 
         if (current.getParentNode() != null) {
             if (current == current.getParentNode().getLeftNode()) {
```

Comment:
have a variable for `current.getParentNode()` before this line
parent node is being repeatedly used.

Same for the other rotation